### PR TITLE
docs: add AGENT.md with AI coding agent guidance

### PR DESCRIPTION
**Pull request by adriantam**
#### What problem is being solved?
Add comprehensive documentation for AI coding agents including architecture overview, key packages, testing guidelines, and maintenance instructions. Highlights reverseexpand.go as critical for ListObjects and includes guideline to review matrix tests when modifying internal/graph or reverseexpand logic.

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

## Summary by CodeRabbit

* **Documentation**
  * Added new documentation file with build/dev commands, architecture overview, package details, API references, port mappings, testing guidelines, and code style conventions.

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# AGENT.md
```
Comment (adriantam) [inline AGENTS.md]: Change this to AGENTS.md

File: AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+```
+HTTP/gRPC Request → Middleware → Server Handler → Command → Graph Resolution → Storage → Response
+```
+
+### Key Packages
+
+**pkg/server/** - gRPC/HTTP server implementation
+- `server.go` - Main server struct implementing OpenFGAServiceServer
+- `commands/` - Business logic for each API (Check, Write, ListObjects, etc.)
+- `commands/reverseexpand.go` - Core ListObjects implementation using reverse expansion
```
Comment (adriantam) [inline AGENTS.md]:
```suggestion
- `commands/reverseexpand/` - Core ListObjects implementation using reverse expansion
```

File: AGENTS.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+Keep this file up to date when making significant changes to the codebase:
+
+- **New packages**: Add to Key Packages section if architecturally significant
+- **New APIs**: Document in Key APIs section
+- **Build changes**: Update Build & Development Commands
+- **Port changes**: Update Ports section
+- **New test patterns**: Add to Testing Guidelines
+- **Import conventions**: Update Code Style if new aliases are adopted
+- **Graph/ListObjects changes**: When modifying `internal/graph/` or `server/commands/reverseexpand.go`, review and stress test matrix tests in `tests/check/` and `tests/listobjects/` to ensure edge cases are covered
```
Comment (adriantam) [inline AGENTS.md]:
```suggestion
- **Graph/ListObjects changes**: When modifying `internal/graph/` or `server/commands/reverseexpand/`, review and stress test matrix tests in `tests/check/` and `tests/listobjects/` to ensure edge cases are covered
```

**Review by DanCech — APPROVED** [review]
Nice
